File: WebApi_CapstoneProject/Document Sharing System/Models/DTOs/UserRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace DSS.Models.DTOs
{
    public class UserRequestDto
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string UserEmail { get; set; } = string.Empty;
        public string UserUsername { get; set; } = string.Empty;
        public Guid DocumentId { get; set; }
        public string DocumentFileName { get; set; } = string.Empty;
        public string RequestType { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public string Reason { get; set; } = string.Empty;
        public DateTime RequestedAt { get; set; }
        public DateTime? ProcessedAt { get; set; }
        public DateTime? AccessGrantedAt { get; set; }
        public DateTime? AccessExpiresAt { get; set; }
        public int? AccessDurationHours { get; set; }
    }

    public class CreateUserRequestDto
    {
        [Required]
        public Guid DocumentId { get; set; }
        
        [Required]
        [StringLength(50)]
        public string RequestType { get; set; } = "Temporary";
        
        [Required]
        [StringLength(500)]
        public string Reason { get; set; } = string.Empty;
        
        [Range(1, 168)] // 1 hour to 1 week
        public int? AccessDurationHours { get; set; } = 24;
    }

    public class ProcessUserRequestDto
    {
        public Guid RequestId { get; set; }
        public string Status { get; set; } = string.Empty; // Approved, Rejected
        public int? AccessDurationHours { get; set; } = 24;
    }

    public class DocumentAccessStatusDto
    {
        public Guid DocumentId { get; set; }
        public string DocumentFileName { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public DateTime? AccessExpiresAt { get; set; }
        public bool HasAccess { get; set; }
        public bool IsExpired { get; set; }
    }
} 

File: Angular_Capstone/DSS/src/app/models/userRequestDto.ts
export interface CreateUserRequestDto {
  documentId: string;
  requestType: string;
  reason: string;
  accessDurationHours?: number;
}

export interface UserRequestDto {
  id: string;
  userId: string;
  userEmail: string;
  userUsername: string;
  documentId: string;
  documentFileName: string;
  requestType: string;
  status: string; // "Pending", "Approved", "Rejected"
  reason: string;
  requestedAt: string; // ISO date string
  processedAt?: string; // ISO date string
  accessGrantedAt?: string; // ISO date string
  accessExpiresAt?: string; // ISO date string
  accessDurationHours?: number;
}

export interface ApproveUserRequestDto {
  accessDurationHours?: number;
}

export interface RejectUserRequestDto {
  reason: string;
} 